zerver/migrations/0530_sessions_data_migration.py

```diff
@@ -0,0 +1,108 @@
+# Generated by Django 4.2.11 on 2024-04-17 04:30
+
+from django.contrib.sessions.models import Session
+from django.db import connection, migrations, transaction
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+from psycopg2.sql import SQL, Composable, Identifier
+
+# This migration copies over data from existing sessions table "django_session"
+# to the custom session table "zerver_realmsession" which is introduced in previous migration.
+
+# Since "django_session" is a large table we want to do the insertion as performantly as possible,
+# approach is pretty simple: drop all indexes, copy data, create back the indexes,
+# because creating an index on pre-existing data is quicker than updating it incrementally as each row is loaded.
+
+# Both drop indexes and copy data are done in one transaction
+# (because we can't write to postgres then update its scheme (i.e. create the indexes) in the same transaction),
```

Comment:
Citation?  AFAIK you can insert data and alter the schema (not "scheme") in the same transaction.  In fact, dropping indexes is altering the schema, so by this argument you should need three transactions.

But I believe this is very much doable in one transaction.

Reply:
> In fact, dropping indexes is altering the schema, so by this argument you should need three transactions.

Yes, but the following error

`OperationalError: cannot CREATE INDEX "zerver_realmsession" because it has pending trigger events`

is raised when creating indexes, not when dropping them.